boxsdk/object/base_api_json_object.py

```diff
@@ -0,0 +1,68 @@
+# coding: utf-8
+
+from __future__ import unicode_literals, absolute_import
+import six
+
+from ..util.translator import Translator
+
+
+class BaseAPIJSONObjectMeta(type):
+    """
+    Metaclass for Box API objects. Registers classes so that API responses can be translated to the correct type.
+    Relies on the _item_type field defined on the classes to match the type property of the response json.
+    But the type-class mapping will only be registered if the module of the class is imported.
+    So it's also important to add the module name to __all__ in object/__init__.py.
+    """
+    def __init__(cls, name, bases, attrs):
+        super(BaseAPIJSONObjectMeta, cls).__init__(name, bases, attrs)
+        item_type = attrs.get('_item_type', None)
+        if item_type is not None:
+            Translator().register(item_type, cls)
+
+
+@six.add_metaclass(BaseAPIJSONObjectMeta)
+class BaseAPIJSONObject(object):
+    """Base class containing basic logic shared between true REST objects and other objects (such as an Event)"""
+
+    _item_type = None
+
+    def __init__(self, response_object=None, **kwargs):
+        """
+        :param response_object:
+            A JSON object representing the object returned from a Box API request.
+        :type response_object:
+            :`dict`
+        :param kwargs:
+            Keyword arguments for base class constructors.
+        :type kwargs:
+            :`dict`
+        """
+        self._response_object = response_object or {}
+        self.__dict__.update(self._response_object)
+        super(BaseAPIJSONObject, self).__init__(**kwargs)
+
+    def __getitem__(self, item):
+        """
+        Try to get the attribute from the API response object.
```

Comment:
You need a blank line after the title line. Otherwise, sphinx won't be able to parse this.